natlib/svcd.c

```diff
@@ -183,5 +190,38 @@ static int svcd_init( lua_State *L )
         lua_call(L, 3, 0);
     }
 
+    
+    return 0;
+}
+
+// Lua: storm.n.ndispatch ( pay, srcip, srcport )
+// Dispatches an incoming notification
+
+static int svcd_ndispatch( lua_State *L )
+{
+    // Checks to make sure it is passed 3 parameters
+    if (lua_gettop(L) != 3) return luaL_error(L, "Expected (pay, srcip, srcport)"); 
+
+    size_t parlen;
+    const char* pay = lua_tolstring(L, -1, &parlen); 
+    uint16_t ivkid = lua_tonumber(L, -1); // changes pay parameter to number and returns
+    lua_getglobal(L, "SVCD"); // gets table
+    lua_pushstring(L, "oursubs"); // use oursubs as the key
+    lua_gettable(L, 5);
+    char ivkidstr[16];
+    sprintf(ivkidstr, "%d", ivkid);
+    lua_pushstring(L, ivkidstr);
+    lua_gettable(L, 6);
+
+    size_t size;
+    const char* item = lua_tolstring(L, 6, &size);
+    if (!lua_isnil(L, 7)) {
+	char newstr[3];
+	strncpy(newstr, pay, 3);
+        lua_pushstring(L, item);
+        lua_pushstring(L, newstr); // pay
+        lua_call(L, 1, 0);
```

Comment:
The function takes one argument only and you are pushing two strings onto the stack before running it. 

Untitled